### PR TITLE
added os_aggregate module

### PR DESCRIPTION
**Pull request by kuboj**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

os_aggregate
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### SUMMARY

This PR adds new module for OpenStack - `os_aggregate` - for managing host aggregates.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @kuboj. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_aggregate.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                if metadata:
+                    cloud.set_aggregate_metadata(aggregate.id, metadata)
+                changed = True
+            else:
+                if _needs_update(module, aggregate):
+                    if availability_zone is not None:
+                        aggregate = cloud.update_aggregate(aggregate.id,
+                            name=name, availability_zone=availability_zone)
+                    if metadata is not None:
+                        metas = metadata
```
Comment (emonty) [inline cloud/openstack/os_aggregate.py:166]: This seems like a leftover from something? metas is only used 3 lines down ...

File: cloud/openstack/os_aggregate.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                             "the shade library MUST be >=1.6.0")
+
+    name = module.params['name']
+    metadata = module.params['metadata']
+    availability_zone = module.params['availability_zone']
+    hosts = module.params['hosts']
+    state = module.params['state']
+
+    if metadata is not None:
+        metadata.pop('availability_zone', None)
```
Comment (emonty) [inline cloud/openstack/os_aggregate.py:134]: Does putting availability_zone in the metadata conflict with the ansible module parameter then? If so, should we toss an error if the user tries to supply one? (For that matter, should we update shade's set_aggregate_metadata to validate and throw an error?)

**Comment by emonty** [comment]
This looks generally great - however, could we rename it to os_nova_host_aggregate? I know that's long - but the current convention is that things that are for Operators and not End Users carry the service name in them. Also, people refer to this feature as "host aggregate" when talking, so I think it would be more discoverable by people if it we don't drop the host.

**Comment by haad** [comment]
this seems to duplicate #2257

**Comment by Shrews** [comment]
As @haad pointed out, there is an existing PR for this. Please coordinate with that author to converge on one PR.

**Pull request closed without merging**